cmd/pydeps.go

```
@@ -66,12 +81,90 @@ func runCheckPyDeps(cmd *cobra.Command, args []string) {
 	}
 }
 
+// Check both rpm and pip are installed on the system, to be used by the Pipcheck
+// and that the local chroot exists, or a repoURL is provided.
+func checkSystemRequirements() {
+	for _, tool := range []string{"createrepo_c", "dnf", "pip"} {
+		err := helpers.RunCommandSilent(tool, "--version")
+		helpers.FailIfErr(err)
```

To help with the defer issue, only call FailIfErr from the very top-level function. I think in this case `runCheckPyDeps` and everything else should return an error.